pandas/tests/io/test_rdata.py

```diff
@@ -0,0 +1,565 @@
+from io import BytesIO
+import os
```

Comment:
Can you add a test that writes and then rereads a DF that contains every non-object dtype. What about ExtensionTypes?

Reply:
Will do. I didn't explore ExtensionTypes which may need conversion handling to R data types.